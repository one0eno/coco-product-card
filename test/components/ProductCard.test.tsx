// import * as React from 'react';
// import * as ReactDOM from 'react-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components';
import { product } from '../data/products';

const { act } = renderer;

describe('ProductTitle', () => {
  test('Debe de mostrar el componente Con sus childs', () => {
    //config
    const wrapper = renderer.create(
      <ProductCard product={product}>{() => <h1>Product card</h1>}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
    //result
  });

  test('Debe de incrementar el contador', () => {
    //config
    const wrapper = renderer.create(
      <ProductCard product={product}>
        {({ count, increaseBy }) => (
          <>
            <h1>Product Card</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(1)}>pulsa</button>
          </>
        )}
      </ProductCard>
    );

    let tree = wrapper.toJSON();
    //console.log(tree);
    expect(tree).toMatchSnapshot();

    act(() => {
      (tree as any).children[2].props.onClick();
    });

    tree = wrapper.toJSON();
    expect((tree as any).children[1].children[0]).toBe('1');
    //result
  });
});
