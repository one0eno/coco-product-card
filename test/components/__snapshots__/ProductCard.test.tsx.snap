// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Debe de incrementar el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    pulsa
  </button>
</div>
`;

exports[`ProductTitle Debe de mostrar el componente Con sus childs 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product card
  </h1>
</div>
`;
